### DISCUSSION TITLE
Props in Page component  is Undefined after using GetStaticProps and GetStaticPaths.

### DISCUSSION THREAD

**Discussion by OgLuka** [vercel/next.js#38727] [Help] [answered]
Hi , I am new to NextJs and I am trying to create a Dynamic page route that needs to map through jobs and show additional info whenever someone clicks on them, photo below for understanding :
![Screenshot 2022-07-17 165254](https://user-images.githubusercontent.com/73040859/179404031-f590c37b-fee8-4af0-8893-329ea5d2ffe8.jpg)

I created a dynamic route and used local JSON file to map through jobs I have. I also used GetStaticProps and GetStaticPaths functions. but for some reason, I can't get it to work every time there is some error that I don't understand.

I am trying to fix this last two days and I can't find the solution and I don't understand things I'm doing wrong even though I'm sure I am doing something wrong, I can also add this to the repo if needed, but help from someone who can easily find the problem will make my day thanks in advance.

This is just a fetcher for my JSON :
   `

      export async function loadjobs() {
  
      const res = await fetch('http://localhost:3000/jobss.json')
  
      const data = await res.json()
  
      return data
    }

   `

       These are my GetStaticProps and GetStaticPaths functions:
  `

           //staticprops
               export async function getStaticProps () {
                  
                  const data = await loadjobs()
                  return {
                  props: {
                      Alljob : data,
                  },
                  }
              }
              //
          
              //staticpaths
              export async function getStaticPaths () {
          
                  const data = await loadjobs()
          
                  const paths = data.map(job => {
                      return { params: { path: job.path }}
                  }
                  )
          
                  return {
                      paths,
                      fallback:false
                  }
              }
              //

`

      and the Page itself :
`   

        import styles from '../../styles/careers.module.css';
        import Link from 'next/link'
        import Head from 'next/head'
        import {jobdata} from '/jobs.js'
        import { useRouter } from 'next/router';
        import { loadjobs } from '../../lib/load-jobs'

                //main page fuction
                export default function Alljob(props) {
            
            
                    //using router
                    const router = useRouter();
                    const {path} = router.query;
            
                    //single job
                    const Job = ({ job }) => {
                        return (
            
                            <Link href={`/careers${job.path}`}>
                                    <div className={styles.Job_div}>
                                        <label>0{job.id}</label>
                                        {router.asPath.includes(job.path) ? <h2 className={styles.red}>{job.name}</h2> : <h2>{job.name}</h2>}
                                    </div>
                            </Link>
                        
                        )
                    }
                    //
                    
                    //job list
                    const Jobs = () => props.Alljob.map(job => {
                        return <Job key={job.id}  job = {job} />
                    }
                    )
                    //
            
                    return (
                    
                    <>
                      <Head>
                            <title></title>
                      </Head>
            
                      <div className={styles.careers_main}>
                          
                           <div className={styles.flex_start_center_100}>
            
                                <div className={styles.careers_headline}>
            
                                        <span className={styles.our_headline_first_small}>C01.Open</span>
                                        <span className={styles.our_headline_second_small}>Vacancies</span>
                                        <span className={styles.our_headline_small_small}>CAREERS</span>
            
                                </div>
            
                           </div>
            
                           <div className={styles.flex_center_center_100}>
            
                                <div className={styles.long_hr}></div>
                               
                           </div>
            
                           <div className={styles.flex_center_center_100}>
            
                                <div className={styles.knowlege_flex_div}>
             
                                     <div className={styles.jobs_list_knowledge}>
                                          <Jobs/>
                                     </div>
            
                                     <div className={styles.knowledge_content_scroll}>
                                           
                                          <span></span>
            
                                     </div>
            
                                </div>
            
                          </div>
            
            
                      </div>
                    </>
                    )
            } 
            

`

**Comment by icyJoseph** [comment, marked as answer]
Hi,

What error are you getting anyway? I'd say it is best to share a link to a repository.

I can guess that this is part of the issue:

```js
 const res = await fetch('http://localhost:3000/jobss.json')
```

Why would this work? Who is running localhost:3000? I think you just need to `import jobsData from 'path/to/data.json';`.

Without knowing what error you are seeing, or a reproduction demo this might take a while to fix.

**Reply by OgLuka** [reply]
Hi @icyJoseph , actually i fixed this and i think it works normally, now i moved to new problem. So as i found out next export with Image tag is complicated, i read many posts and docs but whatever i do i still get error that my next export isn't compatable with next image optimization and because this reason i can't push it to production .
I have tryed changing next.config with custom , loader , unoptimized:true , right now i even imported additional libary for images that uses ExternalImages tag but no matter what i do the error stays same.
if u would look at it i will post a repo, error is just happening while building Thanks.

**Reply by icyJoseph** [reply]
What next.js version do you have? I have released a `next export` project using `next/image` and next@12.2.2 and found no issues at all. Please do share.

**Reply by OgLuka** [reply]
@icyJoseph sorry for the wait this is it https://github.com/OgLuka/Brandergate-public

**Reply by icyJoseph** [reply]
Hi,

I am almost 100% sure that the problem is that you have multiple `module.exports` in `next.config.js`, you need to group them all into one...

Every time you declare `module.exports`, you override the previous export...

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  compiler: {
    // Enables the styled-components SWC transform
    styledComponents: true,
  },
  images: {
    loader: "custom",
    unoptimized: true,
    imageSizes: [16, 32, 48, 64, 96, 128, 256, 384],
    deviceSizes: [640, 750, 828, 1080, 1200, 1920, 2048, 3840],
    nextImageExportOptimizer: {
      imageFolderPath: "public",
      exportFolderPath: "out",
      quality: 75,
    },
  },
  env: {
    storePicturesInWEBP: true,
    generateAndUseBlurImages: true,
  },
  async redirects() {
    return [
      {
        source: "/",
        destination: "/theplatform",
        permanent: false,
      },
    ];
  },
  trailingSlash: true,
};

module.exports = nextConfig;
```

**Reply by OgLuka** [reply]
oh my... I thought something was in next.config as next was not able to read from it, guess yes. I didn't know that thanks so much!!!
